### PR TITLE
Update keys to camelCase and support list of extensions

### PR DESCRIPTION
**Pull request by pragnagopa**
Related issue: https://github.com/Azure/azure-functions-host/issues/3214

### CODE REVIEW

**Comment by elprans** [comment]
@pragnagopa Please update the template in `.ci/appveyor_setup_worker.py` as well.

**Comment by elprans** [comment]
@pragnagopa The tests seem to be failing. Should we bump the host version as well?

**Comment by pragnagopa** [comment]
Right. I will send the Dev build from host soon

**Comment by pragnagopa** [comment]
@elprans - This is the dev build with the config changes. https://ci.appveyor.com/project/appsvc/azure-webjobs-sdk-script-y8o14/build/2.0.11983-cxlninww

**Comment by elprans** [comment]
@pragnagopa You just need to update the `WEBHOST_URL`  in `setup.py`

File: setup.py [hunk 1]
```diff
@@ -15,8 +15,8 @@
 
 
 # TODO: change this to something more stable when available.
-WEBHOST_URL = ('https://ci.appveyor.com/api/buildjobs/0exdyn0sak341kea'
-               '/artifacts/Functions.Binaries.2.0.11886-alpha.no-runtime.zip')
+WEBHOST_URL = ('https://ci.appveyor.com/api/buildjobs/vuqgx4may21qxhc2'
+               '/artifacts/Functions.Private.2.0.11985-alpha.no-runtime.zip')
```
Comment (elprans) [inline setup.py]: We need the `Binaries` archive, not `Private`

Comment (elprans) [reply setup.py]: @pragnagopa ^

**Comment by elprans** [comment]
Has the logging configuration changed? I can no longer enable verbose console logging with `host:logger:consoleLoggingMode': 'always'` in the environment.

**Comment by pragnagopa** [comment]
Yea. We have changes around how we load configuration settings. Can you try setting environment variable **consoleLoggingMode** to always

**Comment by elprans** [comment]
@pragnagopa  Have there been other changes?  The host seems to ignore the extensions in the `bin/` folder of the function app.

**Comment by pragnagopa** [comment]
This pr covers breaking changes wrt to language workers.
@fabiocav - any changes in how we load extensions?

**Comment by pragnagopa** [comment]
Synced up with @fabiocav . We do not have a version of the CLI that installs the right version of extensions yet. As a workaround we need to checkin extensions.json file in the bin folder. I will add that file to this PR and send an update soon.

**Comment by elprans** [comment]
To clarify, the test runner *does* copy `extensions.json` to the bin folder of every test app. The extensions are there, but not being loaded.

**Comment by elprans** [comment]
So this will have to wait until a cli release?

**Comment by pragnagopa** [comment]
@elprans  - Here is the generated extensions.json with the latest versions
```
{
  "extensions":[
    { "name": "CosmosDBConfiguration", "typeName":"Microsoft.Azure.WebJobs.Extensions.CosmosDB.CosmosDBConfiguration, Microsoft.Azure.WebJobs.Extensions.CosmosDB, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null"},
    { "name": "EventGridExtensionConfig", "typeName":"Microsoft.Azure.WebJobs.Extensions.EventGrid.EventGridExtensionConfig, Microsoft.Azure.WebJobs.Extensions.EventGrid, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null"},
    { "name": "HttpExtensionConfiguration", "typeName":"Microsoft.Azure.WebJobs.Extensions.Http.HttpExtensionConfiguration, Microsoft.Azure.WebJobs.Extensions.Http, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null"}
  ]
}
```

**Comment by elprans** [comment]
@fabiocav @pragnagopa How soon is the core tools release that is able to generate the correct metadata?  The worker testing infrastructure assumes that `func extension` does the right thing. There is no hardcoded `extensions.json` anywhere.

**Comment by pragnagopa** [comment]
We should have a staging version of CLI soon. Can you point me to how to install the latest CLI bits for CI system?

**Pull request merged**